src/samplers/sampler.jl

```diff
@@ -32,7 +33,7 @@ assume(spl::Void, dist::Distribution, vn::VarName, vi::VarInfo) = begin
   if haskey(vi, vn)
     r = vi[vn]
   else
-    r = rand(dist)
+    r = init(dist)
```

Comment:
Yes that makes sense, but the problem is this `assume()` is called with `spl = nothing` - we don't have the information of sampler. Maybe it's still OK because the the `assume()` with sampler given still uses `rand()` ?